subsys/bluetooth/services/fast_pair/fp_storage/fp_storage_ak.c

```diff
@@ -349,7 +611,41 @@ static int ak_name_gen(char *name, uint8_t index)
 	return 0;
 }
 
-int fp_storage_ak_save(const struct fp_account_key *account_key)
+static struct fp_bond *fp_bond_get(const void *conn_ctx)
+{
+	for (size_t i = 0; i < ARRAY_SIZE(fp_bonds); i++) {
+		if (fp_bonds[i].conn_ctx == conn_ctx) {
+			return &fp_bonds[i];
+		}
+	}
+
+	return NULL;
+}
+
+static int fp_bond_settings_save(struct fp_bond *bond, struct fp_bond *bond_rollback_copy)
```

Comment:
I doubt if this function should handle the rollback. Maybe we could perform the assignment explicitly in caller to keep single responsibility here (this function unfortunately cannot perform complete rollback on its own)?

Reply:
I don't like this idea. We would have more boilerplate code and risk that someone will eventually not perform rollback.